scripts/exportlegends.lua

```diff
@@ -26,12 +26,501 @@ local MAPS = {
     "Diplomacy",
 }
 
+function getItemSubTypeName(itemType, subType)
+    if (dfhack.items.getSubtypeCount(itemType)) <= 0 then
+        return tostring(-1)
+    end
+    local subtypename = dfhack.items.getSubtypeDef(itemType, subType)
+    if (subtypename == nil) then
+        return tostring(-1)
+    else
+        return tostring(subtypename.name):lower()
+    end
+end
+
+function findEntity(id)
+    for k,v in ipairs(df.global.world.entities.all) do
+        if (v.id == id) then
+            return v
+        end
+    end
+    return nil
+end
+
+function table.contains(table, element)
+  for _, value in pairs(table) do
+    if value == element then
+      return true
+    end
+  end
+  return false
+end
+
+function table.containskey(table, key)
+  for value, _ in pairs(table) do
+    if value == key then
+      return true
+    end
+  end
+  return false
+end
+
+--create an extra legends xml with extra data, by Mason11987 for World Viewer
+function export_more_legends_xml()
+    local julian_day = math.floor(df.global.cur_year_tick / 1200) + 1
+    local month = math.floor(julian_day / 28) + 1 --days and months are 1-indexed
+    local day = julian_day % 28 + 1
+    local year_str = string.format('%0'..math.max(5, string.len(''..df.global.cur_year))..'d', df.global.cur_year)
+    local date_str = year_str..string.format('-%02d-%02d', month, day)
+
+    io.output(tostring(df.global.world.cur_savegame.save_dir).."-"..date_str.."-legends_plus.xml")
+
+    io.write ("<?xml version=\"1.0\" encoding='UTF-8'?>".."\n")
```

Comment:
Already fixed in https://github.com/DFHack/dfhack/commit/d14a8b4308799bba63d27527da756081b63fdb3e (just keep it in mind for future changes)